bug: ConfigurationService - HuggingFace トークンマスキングと database_directory 解決ログ不足

## 概要

ConfigurationService で 3 つの動作不備:
1. HuggingFace トークンがログに平文で出力される (マスキングが "api" section かつ "key" を含むキーのみ対象)
2. `get_database_directory` の相対パス解決時にログが出ない
3. `get_database_directory` の例外時フォールバックが実装されていない

## 失敗テスト

1. `tests/unit/test_configuration_service.py::test_huggingface_token_masking_in_logs`
2. `tests/unit/test_configuration_service.py::test_get_database_directory_resolution_logging`
3. `tests/unit/test_configuration_service.py::test_get_database_directory_error_handling`

## エラー内容

### 1. Token masking
\`\`\`
AssertionError: expected 'hf_1***cdef' in log but got 'hf_1234567890abcdef'
\`\`\`

### 2. Resolution logging
\`\`\`
AssertionError: expected 'Resolved relative database_dir' log call
\`\`\`

### 3. Error handling
\`\`\`
AssertionError: mock_logger.error.assert_called_once() - logger.error was not called
\`\`\`

## 根本原因

### 1. `src/lorairo/services/configuration_service.py:65-81`
\`\`\`python
def update_setting(self, section: str, key: str, value: Any):
    if section == "api" and "key" in key.lower():
        masked = self._mask_secret(value)
    # huggingface_token / hf_token はマスクされない
\`\`\`

### 2, 3. `src/lorairo/services/configuration_service.py:169-175`
\`\`\`python
def get_database_directory(self) -> Path:
    db_dir = self.config["directories"]["database_base_dir"]
    path = Path(db_dir)
    if not path.is_absolute():
        path = Path.cwd() / path  # ← 解決ログなし
    return path  # ← 例外時の fallback なし
\`\`\`

## 修正方針

### 1. マスキング対象を拡張
\`\`\`python
SECRET_KEY_PATTERNS = ("key", "token", "secret")
if any(p in key.lower() for p in SECRET_KEY_PATTERNS):
    masked = self._mask_secret(value)
\`\`\`

### 2. 相対パス解決時に INFO ログ
\`\`\`python
if not path.is_absolute():
    resolved = Path.cwd() / path
    logger.info(f"Resolved relative database_dir: {path} -> {resolved}")
    path = resolved
\`\`\`

### 3. 例外時 fallback
\`\`\`python
try:
    ...
except (KeyError, TypeError, ValueError) as e:
    logger.error(f"Failed to resolve database_dir: {e}", exc_info=True)
    return Path.cwd() / "database"
\`\`\`

## 関連ファイル

- `src/lorairo/services/configuration_service.py:65-81, 169-175`
- `tests/unit/test_configuration_service.py:227-238, 280-293, 296-312`

--- a/src/lorairo/services/configuration_service.py
+++ b/src/lorairo/services/configuration_service.py
@@ -10,6 +10,8 @@ from ..utils.log import logger
 class ConfigurationService:
     """アプリケーションの設定読み込み、更新、保存を担当するサービスクラス。"""
 
+    _SECRET_KEY_PATTERNS = ("key", "token", "secret")
+
     def __init__(
         self, config_path: Path | None = None, shared_config: dict[str, Any] | None = None
     ) -> None:
@@ -73,8 +75,8 @@ class ConfigurationService:
         if section not in self._config:
             self._config[section] = {}
         self._config[section][key] = value
-        # APIキーの場合はマスキングしてログ出力
-        if section == "api" and "key" in key.lower():
+        # キー名に機密情報パターンが含まれる場合はマスキングしてログ出力
+        if any(p in key.lower() for p in self._SECRET_KEY_PATTERNS):
             masked_value = self._mask_api_key(str(value))
             logger.debug("設定値を更新しました: [{}] {} = {}", section, key, masked_value)
         else:
@@ -168,11 +170,22 @@ class ConfigurationService:
 
     def get_database_directory(self) -> Path:
         """directories.database_dir の設定値を取得します。"""
-        dir_str = self.get_setting("directories", "database_dir", "database")
-        # 空文字列の場合もデフォルト値を使用
-        if not dir_str:
-            dir_str = "database"
-        return Path(dir_str).resolve()
+        try:
+            dir_str = self.get_setting("directories", "database_dir", "database")
+            # 空文字列の場合もデフォルト値を使用
+            if not dir_str:
+                dir_str = "database"
+            path = Path(dir_str)
+            if not path.is_absolute():
+                resolved = path.resolve()
+                logger.debug(f"Resolved relative database_dir: {path} -> {resolved}")
+                path = resolved
+            return path
+        except Exception as e:
+            logger.error(f"Failed to resolve database_dir: {e}", exc_info=True)
+            fallback = Path.cwd() / "database"
+            logger.warning(f"Using fallback database directory: {fallback}")
+            return fallback
 
     def get_batch_results_directory(self) -> Path:
         """directories.batch_results_dir の設定値を取得します。"""

